### PR TITLE
Support retrying HTTP requests for remote server errors

### PR DESCRIPTION
**Pull request by lmars**
This will be used like:

```go
opts := tuf.HTTPRemoteOptions{
       UserAgent:          fmt.Sprintf("flynn-host/%s %s-%s %s", version.String(), runtime.GOOS, runtime.GOARCH, name),
       ServerErrorRetries: tuf.DefaultServerErrorRetries,
}
```

### CODE REVIEW

File: client/remote_store.go [hunk 1]
```diff
@@ -51,7 +63,17 @@ func (h *httpRemoteStore) get(s string) (io.ReadCloser, int64, error) {
 	if h.opts.UserAgent != "" {
 		req.Header.Set("User-Agent", h.opts.UserAgent)
 	}
-	res, err := http.DefaultClient.Do(req)
+	var res *http.Response
+	if r := h.opts.ServerErrorRetries; r != nil {
+		for start := time.Now(); time.Since(start) < r.Total; time.Sleep(r.Delay) {
+			res, err = http.DefaultClient.Do(req)
+			if err != nil || res.StatusCode < 500 || res.StatusCode > 599 {
```
Comment (titanous) [inline client/remote_store.go]: Should we retry if there is an `err`?

Comment (lmars) [inline client/remote_store.go]: Yes that probably makes sense, I'll update.

**Comment by lmars** [comment]
@titanous comments addressed

File: client/remote_store.go [hunk 2]
```diff
@@ -8,12 +8,24 @@ import (
...
 )
 
 type HTTPRemoteOptions struct {
-	MetadataPath string
-	TargetsPath  string
-	UserAgent    string
+	MetadataPath       string
+	TargetsPath        string
+	UserAgent          string
+	ServerErrorRetries *HTTPRemoteRetries
```
Comment (titanous) [inline client/remote_store.go]: Maybe just call this `Retries`?

**Comment by titanous** [comment]
LGTM

**Pull request merged**